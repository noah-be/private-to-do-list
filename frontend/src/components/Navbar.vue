<template>
  <nav class="topnav">
    <ul>
      <li><router-link to="/">Home</router-link></li>
      <li><router-link to="/about">About</router-link></li>
      <li><router-link to="/contact">Contact</router-link></li>
    </ul>
  </nav>
</template>

<style lang="scss">

// TODO: Responsive design

@use "@/assets/scss/mixins" as *;

// .topnav {
//   a {
//     color: $secondary-color;
//     text-decoration: none;
//     border: 5px solid $secondary-color;
//     padding: 10px 15px;
//   }

//   ul {
//     list-style-type: none;
//     padding: 0;
//   }

//   li {
//     display: inline;
//     margin-right: 10px;
//   }
// }
</style>
