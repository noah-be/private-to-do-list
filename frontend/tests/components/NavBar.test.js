import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import Navbar from "@/components/Navbar.vue";
import { createRouter, createWebHistory } from "vue-router";

const mockRoutes = [
  { path: "/", name: "home", component: { template: "<div>Home</div>" } },
  {
    path: "/about",
    name: "about",
    component: { template: "<div>About</div>" },
  },
  {
    path: "/contact",
    name: "contact",
    component: { template: "<div>Contact</div>" },
  },
];
const mockRouter = createRouter({
  history: createWebHistory(),
  routes: mockRoutes,
});

describe("Navbar.vue", () => {
  it("renders correctly", () => {
    const wrapper = mount(Navbar, {
      global: {
        plugins: [mockRouter],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("contains the correct navigation links", async () => {
    const wrapper = mount(Navbar, {
      global: {
        plugins: [mockRouter],
      },
    });

    await mockRouter.isReady();

    const links = wrapper.findAll("a");
    expect(links).toHaveLength(3);
    expect(links[0].attributes("href")).toBe("/");
    expect(links[1].attributes("href")).toBe("/about");
    expect(links[2].attributes("href")).toBe("/contact");
  });
});
