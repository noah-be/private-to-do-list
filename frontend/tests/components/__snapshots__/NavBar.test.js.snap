// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar.vue > renders correctly 1`] = `
"<v-container>
  <v-row align="center" justify="center" class="topnav">
    <v-col cols="12" sm="auto" class="d-flex flex-column flex-sm-row">
      <v-btn to="/" text="" router="" class="mx-2 my-2 text-white">Home</v-btn>
      <v-btn to="/about" text="" router="" class="mx-2 my-2 text-white">About</v-btn>
      <v-btn to="/contact" text="" router="" class="mx-2 my-2 text-white">Contact</v-btn>
    </v-col>
  </v-row>
</v-container>"
`;
